pkg/api/v1/events/handlers.go

```diff
@@ -39,228 +49,199 @@ func NewHandler(repository store.Repository, stream events.Stream, logger *logru
 	return &Handler{repository: repository, stream: stream, logger: logger}
 }
 
-func (h *Handler) ackEvent(e *ptypes.StreamEvent) {
-	if err := e.Event.Ack(); err != nil {
+func (h *Handler) ackEvent(m events.Message) {
+	if err := m.Ack(); err != nil {
 		h.logger.WithError(err).Warn("event ack error")
 	}
 }
 
-func (h *Handler) nakEvent(e *ptypes.StreamEvent) {
-	if err := e.Event.Nak(); err != nil {
+func (h *Handler) nakEvent(m events.Message) {
+	if err := m.Nak(); err != nil {
 		h.logger.WithError(err).Warn("event nak error")
 	}
 }
 
 // ControllerEvent handles events from controllers
-func (h *Handler) ControllerEvent(ctx context.Context, streamEvent *ptypes.StreamEvent) {
+func (h *Handler) ControllerEvent(ctx context.Context, msg events.Message) {
 	_, span := otel.Tracer(pkgName).Start(ctx, "ControllerEvent")
 	defer span.End()
 
+	subject := msg.Subject()
+	fragment := strings.TrimPrefix(subject, controllerResponsePrefix)
+	//nolint:gomnd // useless, generic optinion
+	frags := strings.SplitN(fragment, ".", 2)
+
+	// XXX: this would be nicer if we could just destructure the output of SplitN,
+	// but golang is why we can't have nice things
+	facility := frags[0]
+	act := frags[1]
```

Comment:
Worth having a len check on `frags` to make sure its got atleast two parts?

Reply:
yeah, that's a defect on my part.